From 6b57ad06b973cc8ad8263021a4060ac544b107fd Mon Sep 17 00:00:00 2001
From: Angela Stegmaier <angelabaker@ti.com>
Date: Wed, 7 Aug 2019 09:32:16 -0500
Subject: [PATCH 1/2] waylandsink: re-order buffer creation for drm buffers

We should first try to create the wl_buffer using drm api, and
then only if it fails, try dmabuf and shm apis. This patch fixes
the ordering so that the redundant creation does not happen.

Upstream-Status: Pending

Signed-off-by: Angela Stegmaier <angelabaker@ti.com>
---
 ext/wayland/gstwaylandsink.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 19ecf4d..470e9d2 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -772,7 +772,13 @@ gst_wayland_sink_show_frame (GstVideoSink * vsink, GstBuffer * buffer)
       "display, creating it", buffer);
 
   format = GST_VIDEO_INFO_FORMAT (&sink->video_info);
-  if (gst_wl_display_check_format_for_dmabuf (sink->display, format)) {
+
+  if (gst_is_drm_memory (mem)) {
+     wbuf = gst_wl_drm_memory_construct_wl_buffer (mem, sink->display,
+         &sink->video_info);
+  }
+
+  if (!wbuf && gst_wl_display_check_format_for_dmabuf (sink->display, format)) {
     guint i, nb_dmabuf = 0;
 
     for (i = 0; i < gst_buffer_n_memory (buffer); i++)
@@ -783,10 +789,6 @@ gst_wayland_sink_show_frame (GstVideoSink * vsink, GstBuffer * buffer)
       wbuf = gst_wl_linux_dmabuf_construct_wl_buffer (buffer, sink->display,
           &sink->video_info);
   }
-  if (gst_is_drm_memory (mem)) {
-     wbuf = gst_wl_drm_memory_construct_wl_buffer (mem, sink->display,
-         &sink->video_info);
-  }
 
   if (!wbuf && gst_wl_display_check_format_for_shm (sink->display, format)) {
     if (gst_buffer_n_memory (buffer) == 1 && gst_is_fd_memory (mem))
-- 
1.9.1

